src/windows-emulator/windows_emulator.cpp

```diff
@@ -220,6 +220,10 @@ void windows_emulator::setup_process(const application_settings& app_settings, c
     const auto& emu = this->emu();
     auto& context = this->process;
 
+    mod_manager.on_module_load = [this](mapped_module& mod) { this->callbacks().module_loaded(mod); };
+    mod_manager.on_module_unload = [this](mapped_module& mod) { this->callbacks().module_unloaded(mod); };
+    context.on_create_thread = [this](handle h, emulator_thread& thr) { this->callbacks().thread_created(h, thr); };
+
```

Comment:
doing that adds a level of indirection and thus runtime overhead. might not be that important for those specific events, as they don't happen that often, but what about just passing a reference/pointer to the callbacks object into the mod manager?